File: CS/09_Charts/AddShadowEffectForDataLabel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Charts;
using Spire.Presentation.Collections;
using Spire.Presentation.Drawing;

namespace AddShadowEffectForDataLabel
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PowerPoint document.
            Presentation presentation = new Presentation();

            //Load the file from disk.
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Ppt_3.pptx");

            //Get the chart.
            IChart chart = presentation.Slides[0].Shapes[0] as IChart;

            //Add a data label to the first chart series.
            ChartDataLabelCollection dataLabels = chart.Series[0].DataLabels;
            ChartDataLabel Label = dataLabels.Add();
            Label.LabelValueVisible = true;

            //Add outer shadow effect to the data label.
            Label.Effect.OuterShadowEffect = new OuterShadowEffect();

            //Set shadow color.
            Label.Effect.OuterShadowEffect.ColorFormat.Color = Color.Yellow;

            //Set blur.
            Label.Effect.OuterShadowEffect.BlurRadius = 5;

            //Set distance.
            Label.Effect.OuterShadowEffect.Distance = 10;

            //Set angle.
            Label.Effect.OuterShadowEffect.Direction = 90f;         

            String result = "Result-AddShadowEffectToChartDataLabels.pptx";

            //Save to file.
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the PowerPoint file.
            PptDocumentViewer(result);
        }

        private void PptDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/09_Charts/AddShadowEffectForDataLabel.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports Spire.Presentation.Charts
Imports Spire.Presentation.Collections
Imports Spire.Presentation.Drawing

Namespace AddShadowEffectForDataLabel
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PowerPoint document.
			Dim presentation As New Presentation()

			'Load the file from disk.
			presentation.LoadFromFile("..\..\..\..\..\..\Data\Template_Ppt_3.pptx")

			'Get the chart.
			Dim chart As IChart = TryCast(presentation.Slides(0).Shapes(0), IChart)

			'Add a data label to the first chart series.
			Dim dataLabels As ChartDataLabelCollection = chart.Series(0).DataLabels
			Dim Label As ChartDataLabel = dataLabels.Add()
			Label.LabelValueVisible = True

			'Add outer shadow effect to the data label.
			Label.Effect.OuterShadowEffect = New OuterShadowEffect()

			'Set shadow color.
			Label.Effect.OuterShadowEffect.ColorFormat.Color = Color.Yellow

			'Set blur.
			Label.Effect.OuterShadowEffect.BlurRadius = 5

			'Set distance.
			Label.Effect.OuterShadowEffect.Distance = 10

			'Set angle.
			Label.Effect.OuterShadowEffect.Direction = 90f

			Dim result As String = "Result-AddShadowEffectToChartDataLabels.pptx"

			'Save to file.
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the PowerPoint file.
			PptDocumentViewer(result)
		End Sub

		Private Sub PptDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace